types/attributes.pp

```diff
@@ -0,0 +1,14 @@
+# A set of LDAP attributes
+type Openldap::Attributes = Variant[
+  Hash[
+    String[1],
+    Variant[
+      Openldap::Value,
+      Array[Openldap::Value],
+    ],
+  ],
+  Array[
+    Openldap::Key_value,
+  ],
+  Openldap::Key_value,
+]
```

Comment:
Check out stdlib for writing spec tests for types. https://github.com/puppetlabs/puppetlabs-stdlib/blob/main/spec/type_aliases/filemode_spec.rb